src/Microsoft.AspNetCore.Mvc.Core/ProblemDetails.cs

```diff
@@ -18,33 +18,33 @@ public class ProblemDetails
         /// (e.g., using HTML [W3C.REC-html5-20141028]).  When this member is not present, its value is assumed to be
         /// "about:blank".
         /// </summary>
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "type")]
         public string Type { get; set; }
 
         /// <summary>
         /// A short, human-readable summary of the problem type.It SHOULD NOT change from occurrence to occurrence
         /// of the problem, except for purposes of localization(e.g., using proactive content negotiation;
         /// see[RFC7231], Section 3.4).
         /// </summary>
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "title")]
```

Comment:
This may be construed as a breaking change from 2.1. if your app returned `ProblemDetails` and used a Json.NET Contract resolver other than the default. The value specified in `PropertyName` will be used ignoring the naming strategy. 

e.g. If you changed MVC to use `DefaultNamingStrategy` instead of `CamelCase`:
In 2.1: `{ "Title": "Some title" }`
In 2.2: `{ "title": "Some title" }`

Reply:
If Json.NET is used on the client and the JSON is being deserialized then this isn't a problem. Json.NET will fallback to being case-insensitive when it can't find a match.

If JavaScript is used then this could break a user.

Reply:
Discussed with @Eilon and we're OK with the breaking change here due to the rarity of the scenario, the fact that it fixes a bug, and the difficultly of providing a switch for the JSON case.

It would also be possible for someone to work around by defining their own version of ProblemDetails, and using the other features we provide to plug it in.